packages/composables/src/useBreadcrumbs/useBreadcrumbs.ts

```diff
@@ -48,8 +66,38 @@ export function useBreadcrumbs(
     _breadcrumbs.value = [];
   };
 
+  const getCategoryBreadcrumbs = async (categoryId: string) => {
+    const response = await apiClient.invoke(
+      "readBreadcrumb get /breadcrumb/{id}",
+      {
+        pathParams: {
+          id: categoryId,
+        },
+      },
+    );
+    return response.data;
+  };
+
+  const buildDynamicBreadcrumbs = async (categoryId: string) => {
+    try {
+      const breadcrumbs = await getCategoryBreadcrumbs(categoryId);
+
+      _breadcrumbs.value = breadcrumbs.breadcrumbs.map((breadcrumb) => {
+        // Adjust path to be compatible with the router
+        return {
+          ...breadcrumb,
+          path: `/${breadcrumb.path}`,
+        };
+      });
+    } catch (error) {
+      console.error("Error while fetching breadcrumbs", error);
+    }
```

Comment:
we should not handle errors in composable, it should be thrown to invoking component to handle in the project - analytics/notifications etc. however they need to handle it